portable/GCC/RISC-V/portASM.S

```diff
@@ -206,7 +206,12 @@ pxPortInitialiseStack:
     store_x t0, 0(a0)                   /* mstatus onto the stack. */
     addi a0, a0, -portWORD_SIZE         /* Space for critical nesting count. */
     store_x x0, 0(a0)                   /* Critical nesting count starts at 0 for every task. */
+
+#ifdef __riscv_32e
```

Comment:
Looking at this page: https://github.com/riscv-non-isa/riscv-c-api-doc/blob/master/riscv-c-api.md - would using __riscv_e provide better future and wider compatibility than __riscv_32e?

Reply:
Regarding GCC, `__riscv_32e` was added by commit `09baee1ab152` on 2018-05-18, `__riscv_e` by commit `e3354b6de7a3` on 2020-12-11. Both are present at the moment.

Regarding clang, there is no support for RV32E in upstream yet, but the proposed patch adds both of them: https://reviews.llvm.org/D70401

There is nothing about deprecation of `__riscv_32e` in the document. So I would say `__riscv_32e` is currently wider supported.